File: 0125 Valid Palindrome/palindrom.go
func isAlphaNum(b byte) bool {
    return (b >= 'a' && b <= 'z') || (b >= '0' && b <= '9')
}

func isPalindrome(s string) bool {

    left := 0
    right := len(s) - 1

    s1 := strings.ToLower(s)

    for left < right {
        for left < right && !isAlphaNum(s1[left])  {
            left += 1
        }
        for right > left && !isAlphaNum(s1[right]) {
            right -= 1
        }
        if left > right {
            return true
        }
        if s1[left] != s1[right] {
            return false
        }
        left += 1
        right -= 1
    }

    return true
}

File: 0125 Valid Palindrome/palindrom.ts
function isPalindrome(s: string): boolean {

    const clean: string[] = []

    for(const ch of s) {
        if (/[a-zA-Z0-9]/.test(ch) == true) {
            clean.push(ch.toLowerCase());
        }
    }
    console.log(clean)
    var left = 0;
    var right = clean.length - 1;

    while(left <= right) {
        if (clean[left] != clean[right]) {
            return false
        }
        left += 1
        right -= 1
    }
    return true
};

console.log(isPalindrome("race a car"))
console.log(isPalindrome("A man, a plan, a canal: Panama"))